mozilla/fx-private-relay
# Reply to replies mpp 2622

# Forward first free reply and tell user future replies are premium
Right now, when a free user tries to reply to an email, we don't send their reply, because it's a premium feature. But we also DON'T tell the user that their reply was NOT sent.

With this change, we forward a free user's *first* reply, and tell them that future replies require premium.

Figma: https://www.figma.com/file/sSuk3908PyesNG4KkhJIgF/Explorations?node-id=0%3A1&t=HzOxFGvY5kZJ44Ww-0

# Still TODO
* [x] Fix subject of email
* [x] Update the template for the `text/plain` reply
* [x] Make sure all strings are internationalized and ready for translation
* [x] Add `?utm_*` url params to the email's links so we can measure their effectiveness

# Screenshots
## First reply
<img width="926" alt="image" src="https://user-images.githubusercontent.com/71928/209222415-eb5d4a06-1d1a-464f-bef3-07ab044bc4d7.png">

## Next replies
<img width="929" alt="image" src="https://user-images.githubusercontent.com/71928/209222554-850b2e17-f681-40a7-abdd-6a7f635d0e25.png">

# How to test
0. You can use http://127.0.0.1:8000/emails/reply_requires_premium_test to preview the HTML template
   * You can also pass email context vars as url params. E.g., http://127.0.0.1:8000/emails/reply_requires_premium_test?forwarded=True
1. Set up end-to-end email delivery with your local Relay on this branch
2. Send an email to an email alias of a free user
3. In the free user's email, reply to the email from step 2
   * [ ] Verify the reply is sent to the original sender
   * [ ] Verify an email is also sent to the free Relay user telling them that their first reply was sent, but they will need to upgrade to premium to send future replies (see screenshots & figma)
4. In the free user's email, send ANOTHER reply to the email from step 2
   * [ ] Verify this 2nd reply is NOT sent to the original sender
   * [ ] Verify an email is also sent to the free Relay user telling them that they will need to upgrade to premium to send replies (see screenshots & figma)


# Checklist (Definition of Done)
~- [ ] I've added or updated relevant docs in the docs/ directory~
~- [ ] All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).~
